client/settings/payment-settings/promotional-banner/oc-promotion-banner.js

```diff
@@ -0,0 +1,141 @@
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { React } from 'react';
+import styled from '@emotion/styled';
+import apiFetch from '@wordpress/api-fetch';
+import interpolateComponents from 'interpolate-components';
+import CardBody from 'wcstripe/settings/card-body';
+import illustration from 'wcstripe/settings/payment-settings/promotional-banner/illustrations/oc.svg';
+import {
+	BannerIllustration,
+	ButtonsRow,
+	CardColumn,
+	CardInner,
+	DismissButton,
+	MainCTALink,
+} from 'wcstripe/settings/payment-settings/promotional-banner/banner-layout';
+
+const BannerIllustrationBNPL = styled( BannerIllustration )`
+	@media ( min-width: 600px ) {
+		margin: 0 0 -40px 24px;
+	}
+`;
+
+const ButtonsRowBNPL = styled( ButtonsRow )`
+	@media ( min-width: 600px ) {
+		margin-bottom: 0.7em;
+	}
+`;
+
+const ColumnIllustration = styled( CardColumn )`
+	@media ( max-width: 599px ) {
+		text-align: center;
+	}
+`;
+
+const TitleBNPL = styled.h4`
+	margin-top: 0.6em !important;
+	font-weight: 500;
+`;
+
+export const OCPromotionBanner = ( {
+	setShowPromotionalBanner,
+	setIsOCEnabled,
+} ) => {
+	const { createErrorNotice, createSuccessNotice } = useDispatch(
+		'core/notices'
+	);
+
+	const handleBannerDismiss = () => {
+		apiFetch( {
+			path: '/wc/v3/wc_stripe/settings/notice',
+			method: 'POST',
+			data: { wc_stripe_show_oc_promotion_banner: 'no' },
+		} ).finally( () => {
+			setShowPromotionalBanner( false );
+		} );
+		window.location.reload();
```

Comment:
```suggestion
```

I don't think we need to reload the page after dismissing the notice, and if we do, we should do it inside the `finally` block; otherwise, the page might reload with the old value

https://github.com/user-attachments/assets/38222822-aea9-45bb-9307-76d954d2257a